spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/SpringDependencyInjector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.spring;
+
+import static java.util.Objects.requireNonNull;
+
+import org.springframework.beans.factory.BeanFactory;
+
+import com.linecorp.armeria.server.DependencyInjector;
+
+final class SpringDependencyInjector implements DependencyInjector {
+
+    static DependencyInjector of(BeanFactory beanFactory) {
+        requireNonNull(beanFactory, "beanFactory");
+        return new SpringDependencyInjector(beanFactory);
+    }
+
+    private final BeanFactory beanFactory;
+
+    private SpringDependencyInjector(BeanFactory beanFactory) {
+        this.beanFactory = beanFactory;
+    }
+
+    @Override
+    public <T> T getInstance(Class<T> type) {
+        try {
+            return beanFactory.getBean(type);
```

Comment:
Thanks, that's a good suggestion. 👍 
However, can I just leave it as a TODO and implement it later because:
- We haven't had such demands so far
- It increases the complexity of the implementation
- Users can easily apply the different instances using `@DecoratorFactory`

Reply:
Makes sense. We can add the feature later without hurting the original API.

Reply:
Would naming a bean work for other DI frameworks by the way? (Guice, Dagger, etc)

Reply:
Guice and Dagger supports `@Named` bean injection as well.
https://github.com/google/guice/wiki/BindingAnnotations#named